### PR TITLE
UI - new backends

### PR DESCRIPTION
**Pull request by meirish**
Adds support for enabling the new GCP secrets engine and enabling and configuring the new Azure auth method.

screen shot 2018-04-06 at 4 53 26 pm
screen shot 2018-04-06 at 4 53 41 pm
screen shot 2018-04-06 at 4 53 59 pm
screen shot 2018-04-06 at 4 58 11 pm

### CODE REVIEW

**Comment by chrishoffman** [comment]
I wonder if we should put Client Id and Client Secret on the default page.  I think most people are going to use them because it bring better management around authorization to this backend.

**Comment by meirish** [comment]
I think it's still pretty easy to fill out the others - the pattern we've been following is pulling out required ones. Will ping @joshuaogle and get his take too.

File: ui/app/models/auth-config/azure.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import AuthConfig from '../auth-config';
+import fieldToAttrs from 'vault/utils/field-to-attrs';
+
+const { attr } = DS;
+const { computed } = Ember;
+
+export default AuthConfig.extend({
+  tenantId: attr('string', {
+    label: 'Tenant ID',
+    helpText: 'The tenant id for the Azure Active Directory organization',
```
Comment (joshuaogle) [inline ui/app/models/auth-config/azure.js]: Opinion on capitalizing ID? It's `id` in the API, and it's not like ID is a real acronym either, but `id` reads strangely in a regular sentence (comes up below too)

Comment (meirish) [reply ui/app/models/auth-config/azure.js]: Yep, sounds good - added in the latest!

**Pull request merged**